mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvpManager.java

```diff
@@ -56,28 +56,51 @@
 	 * @param id represents the kit ID
 	 * @return the kit from cache or table
 	 */
-	Kit getKit(Transaction transaction, int id) {
-		return existingKits.get(id, num -> {
-			var jooq = transaction.getProperty(DSLContext.class);
+	Kit getKitById(Transaction transaction, int id) {
+		var jooq = transaction.getProperty(DSLContext.class);
 
-			String kitName = jooq
-					.select(KITPVP_KITS_IDS.KIT_NAME).from(KITPVP_KITS_IDS)
-					.where(KITPVP_KITS_IDS.KIT_ID.eq(id)).fetchOne(KITPVP_KITS_IDS.KIT_NAME);
+		String kitName = jooq
+				.select(KITPVP_KITS_IDS.KIT_NAME).from(KITPVP_KITS_IDS)
+				.where(KITPVP_KITS_IDS.KIT_ID.eq(id)).fetchOne(KITPVP_KITS_IDS.KIT_NAME);
 
-			if (kitName == null) {
-				throw new IllegalStateException("Kit by id " + id + " does not exist");
-			}
+		if (kitName == null) {
+			throw new IllegalStateException("Kit by id " + id + " does not exist");
+		}
 
-			DataType<KitItem> itemType = kitItemDataType();
-			Field<KitItem> itemColumn = DSL.field(KITPVP_KITS_CONTENTS.ITEM.getName(), itemType);
-			Set<ItemInSlot> contents = jooq
-                    .select(KITPVP_KITS_CONTENTS.SLOT, itemColumn).from(KITPVP_KITS_CONTENTS)
-                    .where(KITPVP_KITS_CONTENTS.KIT_ID.eq(id)).fetchSet((itemInSlotRecord) -> {
-                        return new ItemInSlot(
-                        		itemInSlotRecord.value1(), itemInSlotRecord.value2());
-                    });
-			return new Kit(id, kitName, Set.copyOf(contents));
-		});
+		return existingKits.get(new KitCacheKey(id, kitName), key -> getKit(transaction, id, kitName));
+	}
+
+	/**
+	 * Gets a kit based on it's kit ID
+	 *
+	 * @param transaction the transaction
+	 * @param name represents the kit ID
+	 * @return the kit from cache or table
+	 */
+	Kit getKitByName(Transaction transaction, String name) {
+		var jooq = transaction.getProperty(DSLContext.class);
+
+		Integer kitId = jooq
+				.select(KITPVP_KITS_IDS.KIT_ID).from(KITPVP_KITS_IDS)
+				.where(KITPVP_KITS_IDS.KIT_NAME.eq(name)).fetchOne(KITPVP_KITS_IDS.KIT_ID);
+
+		if (kitId == null) {
+			throw new IllegalStateException("Kit by name " + name + " does not exist");
+		}
+
+		return existingKits.get(new KitCacheKey(kitId, name), key -> getKit(transaction, kitId, name));
+	}
+
+	Kit getKit(Transaction transaction, int id, String name){
+		var jooq = transaction.getProperty(DSLContext.class);
+
+		DataType<KitItem> itemType = kitItemDataType();
+		Field<KitItem> itemColumn = DSL.field(KITPVP_KITS_CONTENTS.ITEM.getName(), itemType);
+		Set<ItemInSlot> contents = jooq
+				.select(KITPVP_KITS_CONTENTS.SLOT, itemColumn).from(KITPVP_KITS_CONTENTS)
```

Comment:
```suggestion
				.select(KITPVP_KITS_CONTENTS.SLOT, itemColumn)
				.from(KITPVP_KITS_CONTENTS)
```
In SQL, and in Java with JOOQ, it's best to put parts of a query on new lines. The equivalent SQL would be:
```sql
SELECT slot, item
FROM kitpvp_kits_contents
WHERE kit_id = ?
```